### PR TITLE
chore: Test converter for TPF

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Test converter for TPF cluster resource. At the moment it works for `replication_specs`, follow-up PRs will include `advanced_configuration`, `bi_connector_config`, `labels` and `tags`.

Cluster import tests are improved so they can work with both current and TPF cluster.

Link to any related issue(s): CLOUDP-286660

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/advancedcluster/resource_advanced_cluster_test.go [hunk 1]
```diff
@@ -42,19 +41,14 @@ func TestAccClusterAdvancedCluster_basicTenant(t *testing.T) {
 		CheckDestroy:             acc.CheckDestroyCluster,
 		Steps: []resource.TestStep{
 			{
-				Config: configTenant(projectID, clusterName),
+				Config: acc.ConvertAdvancedClusterToTPF(t, configTenant(projectID, clusterName)),
```
Comment (lantoli) [inline internal/service/advancedcluster/resource_advanced_cluster_test.go:44]: example of use of the TPF converter

File: internal/service/advancedcluster/resource_advanced_cluster_test.go [hunk 2]
```diff
@@ -42,19 +41,14 @@ func TestAccClusterAdvancedCluster_basicTenant(t *testing.T) {
...
+				Config: acc.ConvertAdvancedClusterToTPF(t, configTenant(projectID, clusterNameUpdated)),
 				Check:  checkTenant(projectID, clusterNameUpdated),
 			},
-			{
-				ResourceName:      resourceName,
-				ImportStateIdFunc: acc.ImportStateClusterIDFunc(resourceName),
-				ImportState:       true,
-				ImportStateVerify: true,
-			},
+			acc.TestStepImportCluster(resourceName),
```
Comment (lantoli) [inline internal/service/advancedcluster/resource_advanced_cluster_test.go:51]: make import step work in both current and TFP resources, and simplify it

File: internal/testutil/acc/tpf_converter_test.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	"testing"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+)
+
+func TestConvertAdvancedClusterToTPF(t *testing.T) {
+	var (
+		input = `
+			resource "mongodbatlas_advanced_cluster" "cluster2" {
+				project_id   = "MY-PROJECT-ID"
```
Comment (EspenAlbert) [inline internal/testutil/acc/tpf_converter_test.go:13]: Are we covering all blocks?
Maybe missing `bi_connector_config`, `tags`, `labels`, etc.?

**Review by EspenAlbert — APPROVED** [review]
Great! Looking forward to the full block coverage :D

**Pull request merged**